torch/_utils_internal.py

```diff
@@ -53,5 +53,5 @@ def get_source_lines_and_file(obj):
 
 
 TEST_MASTER_ADDR = '127.0.0.1'
-TEST_MASTER_PORT = 29500
+TEST_MASTER_PORT = 0
```

Comment:
I don't think just setting MASTER_PORT to 0 here would work. We need to pass port 0 into TCPStore and then query TCPStore for the port that it actually assigned. After that we need to pass that port number in places like this: https://github.com/pytorch/pytorch/blob/908eee558385461902e9e163db417896732647cf/test/distributed/test_distributed.py#L2059